Dockerfile

```diff
@@ -205,6 +205,22 @@ RUN if [ ! -z "${PG_LOGERRORS}" ]; then \
             && make clean && PG_CONFIG=/usr/lib/postgresql/${pg}/bin/pg_config make install || exit 1 ; \
         done; \
     fi
+
+# Allow Adding Extensions allows *new* files to be created, so that extensions can be added to a running container.
+# Existing files are still owned by root and have their sticky bit (the 1 in the 1750 permission mode) set,
+# and therefore cannot be overwritten or removed by the unprivileged (postgres) user.
+# This ensures the following:
+# - libraries and supporting files for extensions that are part of this Docker Image are immutable
```

Comment:
Clever approach. I wonder whether it would fit our goal of upgrading some extension (i.e. timescale) in place: the problem is the control file that has the same name; it will be owned by root and therefore won't allow to be overwritten from inside the container.

I guess we can mitigate that by setting timescale extension ownership to the postgres user.

Reply:
Yeah, I wanted to start with the most restrictions, relaxing them if needed, instead of adding too many privileges from the start